dev/NavigationView/NavigationView.cpp

```diff
@@ -3339,43 +3550,38 @@ void NavigationView::UpdateSelectionForMenuItems()
     //         <NavigationView.MenuItems>
     //              <NavigationViewItem Content = "Collection" IsSelected = "True" / >
     //         </NavigationView.MenuItems>
-    if (!SelectedItem() && !m_shouldIgnoreNextSelectionChange)
+    if (!SelectedItem())
     {
         if (auto menuItems = MenuItems().try_as<winrt::IVector<winrt::IInspectable>>())
         {
+            bool foundFirstSelected = false;
             for (int i = 0; i < static_cast<int>(menuItems.Size()); i++)
             {
                 if (auto item = menuItems.GetAt(i).try_as<winrt::NavigationViewItem>())
                 {
                     if (item.IsSelected())
                     {
-                        auto scopeGuard = gsl::finally([this]()
-                            {
-                                m_shouldIgnoreNextSelectionChange = false;
-                            });
-                        m_shouldIgnoreNextSelectionChange = true;
-                        SelectedItem(item);
-                        break;
+                        if (!foundFirstSelected)
+                        {
+                            auto scopeGuard = gsl::finally([this]()
+                                {
+                                    m_shouldIgnoreNextSelectionChange = false;
+                                });
+                            m_shouldIgnoreNextSelectionChange = true;
+                            SelectedItem(item);
+                            foundFirstSelected = true;
+                        }
+                        else
+                        {
+                            item.IsSelected(false);
+                        }
                     }
                 }
             }
         }
     }
```

Comment:
I think there are two ways you could make this slightly better, use a ranged based for loop, or use two for loops. I'd prefer the range based for loop I think but here is what I mean by two loops:

```
    if (!SelectedItem())
    {
        if (auto menuItems = MenuItems().try_as<winrt::IVector<winrt::IInspectable>>())
        {
            int i = 0;
            for (i; i < static_cast<int>(menuItems.Size()); i++)
            {
                if (auto item = menuItems.GetAt(i).try_as<winrt::NavigationViewItem>())
                {
                    if (item.IsSelected())
                    {
                        auto scopeGuard = gsl::finally([this]()
                            {
                                m_shouldIgnoreNextSelectionChange = false;
                            });
                        m_shouldIgnoreNextSelectionChange = true;
                        SelectedItem(item);
                        Break;
                    }
                }
            }
            // Deselect the remaining items.
            for(i; i < static_cast<int>(menuItems.Size()); i++)
            {
                if (auto item = menuItems.GetAt(i).try_as<winrt::NavigationViewItem>())
                {
                        item.IsSelected(false);
                }
            }
        }
    }
```